core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -447,7 +450,16 @@ private[spark] class Executor(
       }
     }
 
-    val message = Heartbeat(executorId, tasksMetrics.toArray, env.blockManager.blockManagerId)
+    env.blockTransferService.getMemMetrics(this.executorMetrics)
+    val executorMetrics = if (isLocal) {
+      Utils.deserialize[ExecutorMetrics](Utils.serialize(this.executorMetrics))
```

Comment:
whats the point of this?  sorry if we discussed it earlier ... if its just to test the `ExecutorMetrics` really is serializable, that would be better in a test case